src/controllers/userController.ts

```diff
@@ -0,0 +1,160 @@
+import dotenv from "dotenv";
+import jwt from "jsonwebtoken";
+import passport from "passport";
+import { NextFunction } from "express";
+import { Strategy } from "passport-google-oauth2";
+import { Request, Response } from "express-serve-static-core";
+import { userService } from "../services";
+import { BadRequestError } from "../error-handler/custom-errors";
+import { IAuthorizedRequest } from "../data-mappers/request";
+import userToDTO from "../data-mappers/user";
+
+dotenv.config();
+
+passport.use(
+  new Strategy(
+    {
+      clientID: process.env.GOOGLE_CLIENT_ID as string,
+      clientSecret: process.env.GOOGLE_CLIENT_SECRET as string,
+      callbackURL: process.env.CALLBACK_GOOGLE_URL as string,
+    },
+    function (
+      accessToken: string,
+      refreshToken: string,
+      profile: any,
+      done: any
+    ) {
+      return done(null, profile);
+    }
+  )
+);
+
+passport.serializeUser(function (user: any, done: any) {
+  done(null, user);
+});
+
+const registration = async function (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) {
+  try {
+    const { firstName, lastName, email, password } = req.body;
+    const candidate = await userService.checkCandidate(email);
+    if (candidate) {
+      return next(
+        new BadRequestError({
+          message: `User with email ${email} is already exist`,
+        })
+      );
+    }
+    const userData = await userService.registrationUser(
```

Comment:
We should name our functions/methods starting with verb.
```suggestion
    const userData = await userService.registerUser(
```